pkg/reconciler/eventtransform/eventtransform.go

```diff
@@ -146,8 +163,54 @@ func (r *Reconciler) reconcileJsonataTransformationService(ctx context.Context,
 	return nil
 }
 
-func (r *Reconciler) reconcileJsonataTransformationDeployment(ctx context.Context, expression *corev1.ConfigMap, transform *eventing.EventTransform) error {
-	expected := jsonataDeployment(ctx, r.configWatcher, expression, transform)
+func (r *Reconciler) reconcileJsonataTransformationCertificate(ctx context.Context, transform *eventing.EventTransform) (*cmapis.Certificate, error) {
+	if f := feature.FromContext(ctx); !f.IsStrictTransportEncryption() && !f.IsPermissiveTransportEncryption() {
+		return nil, r.deleteJsonataTransformationCertificate(ctx, transform)
+	}
+	expected := jsonataCertificate(ctx, transform)
+
+	curr, err := r.cmCertificateLister.Certificates(expected.GetNamespace()).Get(expected.GetName())
+	if apierrors.IsNotFound(err) {
+		created, err := r.createCertificate(ctx, transform, expected)
+		if err != nil {
+			return nil, err
+		}
+		if !transform.Status.PropagateJsonataCertificateStatus(created.Status) {
+			// Wait for Certificate to become ready before continuing.
+			return nil, controller.NewSkipKey("")
```

Comment:
why the skip key?

should we factor this out for some more generic reuse, e.g into the `certififcates`?

Reply:
Because the certificate is not ready and we need to wait for it to become ready to continue, so we skip and the object will get re-queued when the certificate status changes